server/src/api/post.rs

```diff
@@ -590,55 +588,34 @@ impl Perform for Oper<EditPost> {
       return Err(APIError::err("site_ban").into());
     }
 
+    // Verify that only the creator can edit
+    if user_id != orig_post.creator_id {
+      return Err(APIError::err("no_post_edit_allowed").into());
+    }
```

Comment:
You could easily move these kinds of checks into separate functions, like `check_is_creator(user_id, post.creator_id) -> Result<(), SecurityError>`. Same for the admin/mod/ban checks.

Reply:
IMO it doesn't make sense for this one, it would just move a simple `!=` check into a function... but it maybe would make sense for the mods and admins one tho.

Reply:
Its not much code, but this is used a lot of times so it adds up. Plus if its in a separate function, you could easily do things like adding log statements or disable a check for testing.

Reply:
Okay done.